### PR TITLE
[test] Adapt Paraview check for PE 18.07

### LINKED ISSUE #381: Adapt Paraview test to PE 18.07

### PR DESCRIPTION
**Pull request by teojgo**
* Adapt to new regression test syntax.

* Make a change in `coloredSphere.py` to load different python modules
  depending on the ParaView version.

Closes #381.

### CODE REVIEW

**Comment by teojgo** [comment]
@jenkins-cscs retry all

File: cscs-checks/apps/paraview/paraview_check.py [hunk 1]
```diff
@@ -1,40 +1,32 @@
...
 
         # Uncomment and adjust to load the ParaView module
         self.modules = ['ParaView']
 
-        # Reset sources dir relative to the SCS apps prefix
         self.executable = 'pvbatch'
         self.executable_opts = ['coloredSphere.py']
 
+        # Needed in order to swap from the default version of gcc
+        self.pre_run = ['module swap gcc/6.2.0 gcc/7.1.0']
```
Comment (vkarak) [inline cscs-checks/apps/paraview/paraview_check.py]: This is not too portable. I suggest specifying `gcc/7.1.0` in `self.modules`. Then ReFrame will generate the necessary commands.

Comment (vkarak) [reply cscs-checks/apps/paraview/paraview_check.py]: It seems that sth is wrong with the module command generation. This looks like related to #59. Can you add a note before `pre_run` referring to this issue?

File: cscs-checks/apps/paraview/paraview_check.py [hunk 2]
```diff
@@ -1,40 +1,34 @@
...
         self.modules = ['ParaView']
 
-        # Reset sources dir relative to the SCS apps prefix
         self.executable = 'pvbatch'
         self.executable_opts = ['coloredSphere.py']
 
+        # NOTE: This is needed in order to swap from the default
+        # version of gcc until issue #59 is fixed. Then it should
+        # be moved to the `self.modules` definition.
+        self.pre_run = ['module swap gcc/6.2.0 gcc/7.1.0']
```
Comment (vkarak) [inline cscs-checks/apps/paraview/paraview_check.py]: After merging with master, this workaround should not be needed.

Comment (teojgo) [reply cscs-checks/apps/paraview/paraview_check.py]: Yes I had it ready and tested with #405

**Comment by vkarak** [comment]
@jenkins-cscs retry daint

**Pull request merged**